packages/browser/src/transports/beacon.ts

```diff
@@ -10,13 +9,13 @@ export class BeaconTransport extends BaseTransport {
   /**
    * @inheritDoc
    */
-  public async captureEvent(event: SentryEvent): Promise<SentryResponse> {
-    const data = serialize(event);
+  public async sendEvent(body: string): Promise<SentryResponse> {
+    const result = global.navigator.sendBeacon(this.url, body);
 
-    const result = global.navigator.sendBeacon(this.url, data);
-
-    return {
-      status: result ? Status.Success : Status.Failed,
-    };
+    return this.buffer.add(
```

Comment:
Are we sure that we want to store it in the buffer here instead of doing it somewhere else? This way we are forcing users to write this code in all custom transports, otherwise it will skip the buffering completely.

On the other hand, it gives them control over buffering... 🤔